homeassistant/components/homematicip_cloud/sensor.py

```diff
@@ -396,6 +435,143 @@ def native_value(self) -> float:
+            type_fn=lambda device: device.functional_channel.energyCounterTwoType,
+        ),
+        HmipEsiSensorEntityDescription(
+            key=ESI_TYPE_ENERGY_COUNTER_INPUT_SINGLE_TARIFF,
+            native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+            device_class=SensorDeviceClass.ENERGY,
+            state_class=SensorStateClass.TOTAL_INCREASING,
+            value_fn=lambda device: device.functional_channel.energyCounterThree,
+            exists_fn=lambda channel: channel.energyCounterThreeType
+            != ESI_TYPE_UNKNOWN,
+            type_fn=lambda device: device.functional_channel.energyCounterThreeType,
+        ),
+    ],
+    ESI_CONNECTED_SENSOR_TYPE_LED: [
+        HmipEsiSensorEntityDescription(
+            key=ESI_TYPE_CURRENT_POWER_CONSUMPTION,
+            native_unit_of_measurement=UnitOfPower.WATT,
+            device_class=SensorDeviceClass.POWER,
+            state_class=SensorStateClass.MEASUREMENT,
+            value_fn=lambda device: device.functional_channel.currentPowerConsumption,
+            exists_fn=lambda channel: channel.currentPowerConsumption is not None,
+            type_fn=lambda device: "CurrentPowerConsumption",
+        ),
+        HmipEsiSensorEntityDescription(
+            key=ESI_TYPE_ENERGY_COUNTER_INPUT_SINGLE_TARIFF,
+            native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+            device_class=SensorDeviceClass.ENERGY,
+            state_class=SensorStateClass.TOTAL_INCREASING,
+            value_fn=lambda device: device.functional_channel.energyCounterOne,
+            exists_fn=lambda channel: channel.energyCounterOne is not None,
+            type_fn=lambda device: ESI_TYPE_ENERGY_COUNTER_INPUT_SINGLE_TARIFF,
+        ),
+    ],
+    ESI_CONNECTED_SENSOR_TYPE_GAS: [
+        HmipEsiSensorEntityDescription(
+            key=ESI_TYPE_CURRENT_GAS_FLOW,
+            native_unit_of_measurement=UnitOfVolumeFlowRate.CUBIC_METERS_PER_HOUR,
+            device_class=SensorDeviceClass.VOLUME_FLOW_RATE,
+            state_class=SensorStateClass.MEASUREMENT,
+            value_fn=lambda device: device.functional_channel.currentGasFlow,
+            exists_fn=lambda channel: channel.currentGasFlow is not None,
+            type_fn=lambda device: "CurrentGasFlow",
+        ),
+        HmipEsiSensorEntityDescription(
+            key=ESI_TYPE_CURRENT_GAS_VOLUME,
+            native_unit_of_measurement=UnitOfVolume.CUBIC_METERS,
+            device_class=SensorDeviceClass.VOLUME,
+            state_class=SensorStateClass.TOTAL_INCREASING,
+            value_fn=lambda device: device.functional_channel.gasVolume,
+            exists_fn=lambda channel: channel.gasVolume is not None,
+            type_fn=lambda device: "GasVolume",
+        ),
+        HmipEsiSensorEntityDescription(
+            key=ESI_TYPE_CURRENT_GAS_VOLUME_PER_IMPULE,
+            native_unit_of_measurement=UnitOfVolumeFlowRate.CUBIC_METERS_PER_HOUR,
```

Comment:
One, I don't think it makes sense to expose this value as it's a static configuration that's used by HmIP to convert "number of impulses" (which it doesn't expose) to volume. Without the raw impulse data there's not much we could do with this value.

Two, it's not in m³/h but in m³/impulse.

Reply:
So removing the GasVolumePerImpulse Sensor is okay?

Reply:
I'd say yes. I see no use case for it. It's a user-supplied config value that tells the ESI how to interpret its raw sensor data. People set it once in the app during initial installation; that's it. It certainly isn't a measurement and won't change on its own.

--

Some background on the hardware:

The sensor is a hall sensor (or reed contact, I haven't broken it open to check which one they use) that gets activated by a small magnet on the last digit of the meter dial. The ESI counts those impulses internally and multiplies them with this config value to derive current and total consumption, which it then reports to the server.

The LED variant is the same, just that there it's a photo-resistive diode/resistor/transistor (again, I haven't cracked one open to check which one) that watches a blinking red LED on the meter.

Only the IEC sensor is a bit more involved, as the meter sends its data as a digital signal using an IR LED.

In all three cases, the actual HMIP device is identical, it's just shipped with a different sensor in the box. That's why you get that "everything mixed together" reading.

Reply:
Thx for the input. I removed it.